### PR TITLE
[HDFS] Readiness check for data nodes

### PR DESCRIPTION
**Pull request by NimaVaziri**

### CODE REVIEW

**Comment by triclambert** [comment]
retest this please

File: frameworks/hdfs/src/main/dist/svc.yml [hunk 1]
```diff
@@ -167,6 +167,11 @@ pods:
             dest: hadoop-2.6.0-cdh5.7.1/etc/hadoop/hdfs-site.xml
         env:
           FRAMEWORK_NAME: {{SERVICE_NAME}}
+        readiness-check:
```
Comment (gabrielhartmann) [inline frameworks/hdfs/src/main/dist/svc.yml]: YAML allows comments.  Can you write a quick comment explaining what this readiness check does?

**Comment by gabrielhartmann** [comment]
@NimaVaziri:  Can you add the requested comment and rebase?  Would like to get this in.

**Comment by NimaVaziri** [comment]
Oh the comments are now included, unless you prefer them to be more detailed.

**Comment by NimaVaziri** [comment]
@gabrielhartmann given that the other problem we ran into isn't necessarily related to readiness checks, unless you have more comments, this should be good to go

File: frameworks/hdfs/src/main/dist/svc.yml [hunk 2]
```diff
@@ -158,6 +166,12 @@ pods:
             dest: {{HDFS_VERSION}}/etc/hadoop/hdfs-site.xml
         env:
           FRAMEWORK_NAME: {{SERVICE_NAME}}
+        # If HDFS does not report a data node to be live, we don't consider the data node to be ready yet
+        readiness-check:
+          cmd: IP="$(ifconfig eth0 | grep "inet " | awk '{print $2}')" && MATCH="$(./{{HDFS_VERSION}}/bin/hdfs dfsadmin -report | grep $IP | wc -l)" && [ $MATCH -gt 1 ]
```
Comment (gabrielhartmann) [inline frameworks/hdfs/src/main/dist/svc.yml]: Your IP address can be determined by the `LIBPROCESS_IP` envvar.

File: frameworks/hdfs/universe/config.json [hunk 3]
```diff
@@ -164,6 +164,11 @@
       "type": "object",
       "description": "HDFS File System configuration options",
       "properties": {
+        "version": {
```
Comment (gabrielhartmann) [inline frameworks/hdfs/universe/config.json]: Please don't expose this as configurable.  We don't allow the changing of the underlying service version within the context of a single package version.

**Comment by NimaVaziri** [comment]
@gabrielhartmann addressed comments

**Comment by gabrielhartmann** [comment]
@NimaVaziri:  LGTM.  Can you resolve the merge conflicts please?

**Comment by NimaVaziri** [comment]
retest this please

**Comment by nickbp** [comment]
retest this please

**Pull request merged**